torch/_torch_docs.py

```diff
@@ -2698,6 +2707,8 @@
 
 If :attr:`mat` is a `n x m` Tensor, :attr:`vec` is a 1D Tensor of size `m`, :attr:`out` will be 1D of size `n`.
 
+.. note:: This function does not :ref:`broadcast <broadcasting-semantics>`.
```

Comment:
I agree with your points. I think having explicit mentions to `matmul` where appropriate will simplify the user's effort to remember function names, as it's a single function with intuitive behaviour that is equivalent to numpy's.